core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/AwsExecutionContextBuilder.java

```diff
@@ -135,6 +142,35 @@ private AwsExecutionContextBuilder() {
                                .build();
     }
 
+    private static void putAuthSchemeResolutionAttributes(ExecutionAttributes executionAttributes,
+                                                          SdkClientConfiguration clientConfig) {
+
+        // review TODO: should this be client v/s advanced option?
```

Comment:
>I was thinking changing auth schemes and their providers would be on the generated service client builder.

Currently I haven't implemented any of the entry points for changing these. For auth scheme provider, yeah it should be in the generated service client builder so it can be the specific {Service}AuthSchemeProvider. For putAuthScheme, that would be on the AwsClientBuilder as it is not service specific. Your design doc calls out both these. But maybe it is a typo, but your said "and their provider**s**" (plural), so not sure if you are talking about something else there. If so, please clarify.

Either ways, my _review TODO_ question here is about, whether the client option for AUTH_SCHEME_PROVIDER be SdkClientOption.AUTH_SCHEME_PROVIDER or SdkAdvancedClientOption.AUTH_SCHEME_PROVIDER. 
* Today SIGNER is SdkAdvancedClientOption.SIGNER.
* I guess I'm not clear what should be clientOption v/s advancedClientOption. 
* I see making it advanced client option would make the user do `clientBuilder.overrideConfiguration(b->b.putAdvancedOption(AUTH_SCHEME_RESOLVER, resolver))` v/s clientBuilder.authSchemeResolver(resolver). Well, in this case, since we want to use specific {Service}AuthSchemeProvider, using putAdvancedOption would not by type-safe. So makes sense to not use Advanced here.
* But question applies to AUTH_SCHEMES and IDENTITY_PROVIDER_CONFIGURATION options - they are not service specific, so should they be client v/s advnaced?

Reply:
Hmm, I just noticed that SdkAdvancedClientOption is `@SdkPublicApi` whereas SdkClientOption is `@SdkProtectedApi`! So where something should go still feels a bit confusing, but atleast I know I'll err on the side of not making it public unless I'm sure it needs to be, so I have the answer to my _TODO_ question - will make these all clientOption (not advanced).